src/db/collections/configuration_update.rs

```diff
@@ -64,40 +68,18 @@ impl ConfigurationUpdateCollection {
         ledger_index: MilestoneIndex,
         config: NodeConfiguration,
     ) -> Result<(), Error> {
-        if let Some(latest_config) = self.get_node_configuration_for_ledger_index(ledger_index).await? {
-            if latest_config.ledger_index == ledger_index {
-                if latest_config.config != config {
-                    self.replace_one(doc! {}, ConfigurationUpdateDocument { ledger_index, config }, None)
-                        .await?;
-                }
-            } else {
-                self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                    .await?;
-            }
-        } else {
-            self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                .await?;
-        }
-
-        Ok(())
-    }
-
-    /// Add the node configuration to this collection if it is newer and different.
-    pub async fn update_latest_node_configuration(
-        &self,
-        ledger_index: MilestoneIndex,
-        config: NodeConfiguration,
-    ) -> Result<(), Error> {
-        if let Some(latest_config) = self.get_latest_node_configuration().await? {
-            if latest_config.ledger_index >= ledger_index {
-                return Ok(());
-            } else if latest_config.config != config {
-                self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                    .await?;
-            }
-        } else {
-            self.insert_one(ConfigurationUpdateDocument { ledger_index, config }, None)
-                .await?;
+        let params = self.get_node_configuration_for_ledger_index(ledger_index).await?;
+        if params.is_none()
+            || params
+                .map(|latest_config| latest_config.config != config)
+                .unwrap_or_default()
+        {
+            self.update_one(
+                doc! { "_id": ledger_index },
+                doc! { "$set": mongodb::bson::to_bson(&config).unwrap() },
```

Comment:
Undocumented `unwrap`.

Reply:
This isn't exactly the first one for `to_bson`.